guava/src/com/google/common/escape/UnicodeEscaper.java

```diff
@@ -127,8 +133,10 @@ public String escape(String string) {
    * @throws IllegalArgumentException if the scanned sub-sequence of {@code csq} contains invalid
    *     surrogate pairs
    */
-  protected int nextEscapeIndex(CharSequence csq, int start, int end) {
-    int index = start;
+  @SuppressWarnings("upperbound:compound.assignment.type.incompatible")//index += 2 only if Character.isSupplementaryCodePoint(cp) is true.
+  //This won't happen since loop breaks if `cp` < 0.
```

Comment:
What is "this" referring to in this sentence? As a general rule, it's usually bad style in English to just say "this" - prefer "this event", "this situation", "this person", etc. instead. The reason to do so is to make clear to the reader exactly what the antecedent of the "this" is.

Reply:
As for the actual justification - if it (interpreting "this" to refer to index being incremented by 2) won't happen, why is it in the code?